pkg/event/event.go

```diff
@@ -0,0 +1,123 @@
+package event
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	WorkCompletion              = "work_completion"
+	WorkFailure                 = "worker_failure"
+	WorkDistribution            = "work_distribution"
+	WorkExecutionStart          = "work_execution_start"
+	WorkExecutionTimeout        = "work_execution_timeout"
+	RemoteWorkerConnection      = "remote_work_connection"
+	StreamCreation              = "stream_creation"
+	WorkRequestSerialization    = "work_request_serialized"
+	WorkResponseDeserialization = "work_response_serialized"
+	LocalWorkerFallback         = "local_work_executed"
+)
+
+type Event struct {
+	Name      string
+	Timestamp time.Time `json:"timestamp"`
+	Data      map[string]interface{}
+}
+
+type EventTracker struct {
+	events    []Event
+	mu        sync.Mutex
+	logger    *logrus.Logger
+	config    *Config
+	apiClient *EventClient
+}
+
+func NewEventTracker(config *Config) *EventTracker {
```

Comment:
since this package seems to be self-contained, would benefit from having unit tests

Reply:
yeah I think it should once @restevens402 goes e2e with a first stab at implementation which is 90% done and should be done tomorrow